repos/system_upgrade/el8toel9/actors/firewalldcheckallowzonedrifting/actor.py

```diff
@@ -0,0 +1,52 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.actor import private
+from leapp.models import FirewallsFacts
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class FirewalldCheckAllowZoneDrifting(Actor):
+    """
+    This actor will check if AllowZoneDrifiting=yes in firewalld.conf. This
+    option has been removed in RHEL-9 and behavior is as if
+    AllowZoneDrifiting=no.
+    """
+
+    name = 'firewalld_check_allow_zone_drifting'
+    consumes = (FirewallsFacts,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        conf_key = "AllowZoneDrifting"
+
+        # non existent config means the default (no) will be used
+        #
+        path = private.getPathToFirewalldConf()
+        if not path:
+            return
+
+        # If firewalld is not enabled then don't bother the user about its
+        # configuration. This Report keys off a _default_ value and as such
+        # will trigger for all users that have not done one of the following:
+        #   - disabled firewalld
+        #   - manually set AllowZoneDrifting=no (as firewalld logs suggests)
+        #
+        for facts in self.consume(FirewallsFacts):
+            if not facts.firewalld.enabled:
+                return
+
+        conf_dict = private.read_config(path)
+        if private.isAllowZoneDriftingInUse(conf_dict):
+            create_report([
+                reporting.Title("Unsupported configuration"),
+                reporting.Summary("Firewalld is using configuration option "
+                                  + "'%s' which has been removed in RHEL-9. "
+                                  % (conf_key)
+                                  + "New behavior is as if '%s' was set to 'no'."
+                                  % (conf_key)),
```

Comment:
Please do not use % formatting, but use .format
So to get you an idea:
```python
'Starting with RHEL-9, FirewallD removed the support for the configuration option `{option}`.'.format(option=conf_key)
```

The variable '{option}' can be also used multiple times inside of the string, so no need to repeat the argument. That's the advantage of the named formatting strings.